cclib/parser/molcasparser.py

```diff
@@ -0,0 +1,181 @@
+
+    def before_parsing(self):
+        # Compile the regex for extracting the element symbol from the
+        # atom label in the "Molecular structure info" block.
+        self.re_atomelement = re.compile('([a-zA-Z]+)\d+')   
+
+        # Compile the dashes-and-or-spaces-only regex.
+        self.re_dashes_and_spaces = re.compile('^[\s-]+$')
+
+    def after_parsing(self):
+        pass
+
+    def extract(self, inputfile, line):
+        """Extract information from the file object inputfile."""
+
+        ## This section is present when executing &GATEWAY.
+        # ++    Molecular structure info:
+        #       -------------------------
+
+        #                     ************************************************ 
+        #                     **** Cartesian Coordinates / Bohr, Angstrom **** 
+        #                     ************************************************ 
+
+        #      Center  Label                x              y              z                     x              y              z
+        #         1      C1               0.526628      -2.582937       0.000000              0.278679      -1.366832       0.000000
+        #         2      C2               2.500165      -0.834760       0.000000              1.323030      -0.441736       0.000000
+        if line[25:63] == 'Cartesian Coordinates / Bohr, Angstrom':
+            if not hasattr(self, 'atomnos'):
+                self.atomnos = []
+            if not hasattr(self, 'atomcoords'):
+                self.atomcoords = []
+
+            self.skip_lines(inputfile, ['stars', 'blank', 'header'])
+
+            line = next(inputfile)
+
+            atomelements = []
+            atomcoords = []
+
+            while not self.re_dashes_and_spaces.search(line):
+                sline = line.split()
+                atomelements.append(self.re_atomelement.search(sline[1]).groups()[0])
+                atomcoords.append(list(map(float, sline[5:])))
+                line = next(inputfile)
+
+            self.atomcoords.append(atomcoords)
+
+            if self.atomnos == []:
+                self.atomnos = [utils.PeriodicTable().number[atomelement] for atomelement in atomelements]
+
+            if not hasattr(self, 'natom'):
+                self.set_attribute('natom', len(self.atomnos))
+
+        ## This section is present when executing &SCF.
+        if line[0:29] == '++    Orbital specifications:':
+
+            self.skip_lines(inputfile, ['dashes', 'blank'])
+
+            line = next(inputfile)
+
+            while line[0:2] != '--':
+
+                if line[6:30] == 'Total number of orbitals':
+                    self.set_attribute('nmo', int(line.split()[-1]))
+                if line[6:31] == 'Number of basis functions':
+                    self.set_attribute('nbasis', int(line.split()[-1]))
+
+                line = next(inputfile)
+
+        if line[6:23] == 'Molecular charge ':
+            self.set_attribute('charge', int(float(line.split()[-1])))
```

Comment:
Are you sure this will round how you want it to? Calling `int` will round towards zero.